### DISCUSSION TITLE
how to copy the function

### DISCUSSION THREAD

**Discussion by newminder** [YetiForceCompany/YetiForceCRM#16970] [How to] [answered]
in Calendar, if the account was chose,  then it will **limit the search result** when choosing contact?
how can I copy this limitation function to the Quotes? 
anyone can help?

**Comment by mariuszkrzaczkowski** [comment]
It works in calendar https://gitdeveloper.yetiforce.com/index.php?module=Calendar&view=Calendar&mid=46&parent=44
![image](https://user-images.githubusercontent.com/10198654/226274648-e1dbc3ca-ee68-4027-9dbe-f1fbb4fefd8b.png)
![image](https://user-images.githubusercontent.com/10198654/226274682-279120b3-7ca5-4155-a971-bff0a3fa232f.png)

**Reply by newminder** [reply]
Yes. I want same function in Quotes,  when choosing the contact, the popup result should be limited by Account...how should I do?

**Reply by newminder** [reply]
I tried below codes in the /modules/Contacts/models/Module.php, but not working. 
`if ('SQuotes' === $sourceModule && $record) {
$queryGenerator->addJoin(['INNER JOIN', 'u_#__squotes', 'u_#__squotes.accountid = vtiger_contactdetails.parentid AND u_#__squotes.squotesid = :records', [':records' => $record]]);
}`

**Reply by mariuszkrzaczkowski** [reply]
https://github.com/YetiForceCompany/YetiForceCRM/blob/8b3d0cb0cc12890e05c59f199129e8adbeaab5eb/app_data/moduleHierarchy.php#L58-L61

**Reply by newminder** [reply]
hello Mariusz Krzaczkowski, thanks very much for your patience and time.  I believe your codes is about to relate quotes and account, that's not what I need.  

let me to be more clear,  I have fileds [account] and [contact] in Quotes module. When I creating a quote record, I choose account, then I choose contact but all contacts come out in the popup window,  how should I limit contact according to the account I chose? 

like it works in the calendar,  you choose account, then when you choose contact it will popup the results limited by  account.

**Reply by newminder** [reply]
btw,  after I adding the codes 
`if ('SQuotes' === $sourceModule && $record) { $queryGenerator->addJoin(['INNER JOIN', 'u_#__squotes', 'u_#__squotes.accountid = vtiger_contactdetails.parentid AND u_#__squotes.squotesid = :records', [':records' => $record]]); }`
in the /modules/Contacts/models/Module.php

then I found this ---  when I create a new quote record, I choose the account and then choose contact all the contact come out, but if I leave the contact blank and save the record and re edit this record, then the function of limitation of  contact works, the pop up window just show me the results according to the account. 

but this is very inconvenient.

**Reply by newminder** [reply]
I guess the above the code can get account value from database, but not the input value

**Reply by newminder** [reply]
I test the below record and tried to put a actual value then it works,  now what I need to do is to get the accountid value
`if ('SQuotes' === $sourceModule && $record) {
		  $condition = ['and', ['vtiger_contactdetails.parentid' =>**_how to get the field value_** ]];
		  $queryGenerator->addNativeCondition($condition);
}`

**Comment by newminder** [comment]
anyone knows how to get input field value?

**Comment by apcloic** [comment, marked as answer]
@newminder
At this step, your input value is not saved in the database yet so I believe you need to use javascript.

Also, if you look closely at account field in calendar, it's a referenceLink UIType and not uitype 10 like in Quotes module and many other modules.
https://github.com/YetiForceCompany/YetiForceCRM/blob/2379514058fcfd262ff4adb68b038604f6c57ce0/modules/Vtiger/uitypes/ReferenceLink.php

The good news is that you have a working example of what you want to achieve in Calendar module, but you need to better study how it works and then find how to replicate the function in another module :)

Regards,

**Reply by newminder** [reply]
thanks, I already changed the field uitype from database. but still stuck here.

**Reply by apcloic** [reply]
@newminder 
Nice but there may be other files & functions, like contact field in calendar module also use au specific **referenceExtend** uitype.
Did you also check template files and javascript functions ?
Keep searching and you will find :)
Good luck

**Reply by newminder** [reply]
thanks.  I looked some files under layouts and tried to figure out what's wrong going there.  in the Calendar module, it gets the filed value to post it to the popup window as $related_parent_id,  but in other modules, the field value would not post to the $related_parent_id,   this means I did not trigger the mechanism to past the value to the popup window.
